specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,544 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ComputeDiagnosticResourceProviderClient",
+    "description": "The Compute Diagnostic Resource Provider Client",
+    "version": "2021-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnosticOperations/{operationId}": {
+      "get": {
+        "tags": [
+          "diagnosticOperations"
+        ],
+        "operationId": "DiagnosticOperations_Read",
+        "description": "Gets status of a Diagnostic operation issued for Disk Inspection.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "in": "path",
+            "name": "operationId",
+            "required": true,
+            "type": "string",
+            "description": "The identifier pointing to a request for Disk Inspection. This is used for polling result of the request."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
```

Comment:
The GET requests acts as a polling request. If the diagnostics operation is still ongoing, the response will be 202, if the operation has completed, it will be 200.

Reply:
This isn't the right async model. Neither the current RPC async model (location header + provisioningState) nor the old-style Azure-Operation model allow for 202 from GET.  The GET should always return 200 with operation status payload. Ref: [https://github.com/microsoft/api-guidelines/blob/vNext/graph/Guidelines-deprecated.md#1324-operations-resource](https://github.com/microsoft/api-guidelines/blob/vNext/graph/Guidelines-deprecated.md#1324-operations-resource).

Reply:
Hello @mentat9,
CDRP is a proxy-only location based RP which enables diagnostics on a given resource. CDRP team followed guidelines for **(Location + Retry-After)** model outline in async-operations document when adding swagger for async API ->
POST : /locations/diagnostics/diskInspection/run. The response header will outline location which user can poll on to get status of the operation created as part of POST. "Location" header contains the GET url that customers can poll on.
GET: /locations/diagnosticsOperations/{operationId} will indicate status of accepted {operationId}. This is in accordance with guidelines stated here: https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-resource-manager/management/async-operations.md
<img width="760" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/35848086/92a723e9-7e72-4883-b31a-551424422557">

These diagnostics APIs were reviewed and signed off by ARM-API review twice. These APIs are actively used by internal workloads to run diagnostics and customers opted into public-preview, hence we are not inclined to have breaking API changes.

Reply:
I'm fine with not breaking anything. @mentat9 any remaining concerns?